src/llvm/di.rs

```diff
@@ -348,15 +348,41 @@ unsafe fn iter_operands(v: LLVMValueRef) -> impl Iterator<Item = LLVMValueRef> {
     (0..LLVMGetNumOperands(v)).map(move |i| LLVMGetOperand(v, i as u32))
 }
 
-unsafe fn iter_metadata_copy(v: LLVMValueRef) -> impl Iterator<Item = (u32, LLVMMetadataRef)> {
-    let mut count = 0;
-    let entries = LLVMGlobalCopyAllMetadata(v, &mut count);
-    (0..count).map(move |index| {
-        (
-            LLVMValueMetadataEntriesGetKind(entries, index as u32),
-            LLVMValueMetadataEntriesGetMetadata(entries, index as u32),
-        )
-    })
+struct MetadataEntries {
```

Comment:
Can we move it to `src/llvm/types/di.rs`?

Reply:
Not all metadata is DI?

Reply:
True. I see that the `LLVMGlobalCopyAllMetadata` belongs to `llvm/lib/IR/Core.cpp`. I would then keep it in `src/llvm/types/ir.rs` (or, eventually, but probably too muc effort - split `ir.rs` into a subdir with `mod.rs` and `core.rs`, where eventual wrappers for other `IR/*` files could have their own files).

My main point is that I would prefer to not mix wrappers (which are currently in the `types` subdir) and DI sanitization logic in the same module.